internal/provider/resource_tfe_sentinel_version.go

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceTFESentinelVersion() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFESentinelVersionCreate,
+		Read:   resourceTFESentinelVersionRead,
+		Update: resourceTFESentinelVersionUpdate,
+		Delete: resourceTFESentinelVersionDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceTFESentinelVersionImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"version": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"sha": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"official": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"beta": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"deprecated": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"deprecated_reason": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  nil,
+			},
+		},
+	}
+}
+
+func resourceTFESentinelVersionCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(ConfiguredClient)
+
+	opts := tfe.AdminSentinelVersionCreateOptions{
+		Version:          *tfe.String(d.Get("version").(string)),
+		URL:              *tfe.String(d.Get("url").(string)),
+		SHA:              *tfe.String(d.Get("sha").(string)),
+		Official:         tfe.Bool(d.Get("official").(bool)),
+		Enabled:          tfe.Bool(d.Get("enabled").(bool)),
+		Beta:             tfe.Bool(d.Get("beta").(bool)),
+		Deprecated:       tfe.Bool(d.Get("deprecated").(bool)),
+		DeprecatedReason: tfe.String(d.Get("deprecated_reason").(string)),
+	}
+
+	log.Printf("[DEBUG] Create new Sentinel version: %s", opts.Version)
+	v, err := config.Client.Admin.SentinelVersions.Create(ctx, opts)
+	if err != nil {
+		return fmt.Errorf("Error creating the new Sentinel version %s: %w", opts.Version, err)
+	}
+
+	d.SetId(v.ID)
+
+	return resourceTFESentinelVersionUpdate(d, meta)
```

Comment:
Same here. Can you share more about why this method returns `Update` instead of `Read`?

Reply:
same as above ^ https://github.com/hashicorp/terraform-provider-tfe/pull/1202#discussion_r1445593209